azure-pipelines.yml

```diff
@@ -0,0 +1,90 @@
+parameters:
+- name: module_names
+  type: object
+  default:
+  - nidigital
+  - nitclk  
+  - nifgen
+  - nidcpower
+  - nidmm
+  - niscope
+  - nimodinst
+  - nise
+  - niswitch
+
+# A pipeline with no CI trigger
+trigger: none
+
+# specific path
+pr:
+  branches:
+    include:
+    - master
+    - releases/*
+  paths:
+    exclude:
+    - CHANGELOG.md
+    - CONTRIBUTING.md
+    - .gitattributes
+    - LICENSE
+    - VERSION
+  drafts: false
+
+stages:
+
+  - stage: ensure_code_gen_up_to_date
+    pool:
+      name: nimi-bot
+      demands:
+        - nimi-bot -equals True
+        - agent.os -equals Linux
+    jobs:
+    - job: ensure_code_gen_up_to_date
+      workspace:
+        clean: all
+      steps:
+        - checkout: self  
+          clean: true
+          fetchDepth: $(clone_depth)
+        - task: PythonScript@0
+          displayName: Ensure codegen up to date
+          inputs:
+            scriptSource: inline
+            script: |
+              import os
+              import sys
+              import subprocess
+
+              # To verify code is generated
+              if os.system('git config user.email "dummy@ni.com"') != 0:
+                sys.exit(-1)
```

Comment:
Lots of checks for error conditions but they all return the same code. Help developers debug this by using distinct codes and/or printing a helpful error to stderr